src/mozanalysis/experiment.py

```diff
@@ -367,17 +393,35 @@ def build_enrollments_query(
             time_limits, enrollments_query_type
         )
 
+        if exposure_metric:
+            exposure_query = (
+                custom_exposure_query
+                or exposure_metric.data_source.build_exposure_query(
+                    exposure_metric, time_limits, self.experiment_slug
+                )
+            )
+        else:
+            exposure_query = custom_exposure_query or self._build_exposure_query(
+                time_limits, enrollments_query_type
+            )
+
         segments_query = self._build_segments_query(segment_list, time_limits)
 
         return """
             WITH raw_enrollments AS ({enrollments_query}),
-            segmented_enrollments AS ({segments_query})
+            segmented_enrollments AS ({segments_query}),
+            exposures AS ({exposure_query})
 
-            SELECT se.*
+            SELECT
+                se.*,
+                e.*
             FROM segmented_enrollments se
+            LEFT JOIN exposures e
+            USING (client_id, branch)
         """.format(
             enrollments_query=enrollments_query,
             segments_query=segments_query,
+            exposure_query=exposure_query,
```

Comment:
This would add two columns to the enrollment tables `exposure_date` and `num_exposure_events`. I think having a single table with enrollment and exposure data is slightly more convenient for building the metrics queries rather than having separate enrollment and exposure tables.